lib/AST/Decl.cpp

```diff
@@ -6992,6 +6992,18 @@ Expr *ParamDecl::getTypeCheckedDefaultExpr() const {
   return new (ctx) ErrorExpr(getSourceRange(), ErrorType::get(ctx));
 }
 
+Type ParamDecl::getTypeOfDefaultExpr() const {
+  auto &ctx = getASTContext();
+
+  if (Type type = evaluateOrDefault(
+          ctx.evaluator,
+          DefaultArgumentTypeRequest{const_cast<ParamDecl *>(this)}, nullptr)) {
```

Comment:
Here, can we use `Type()` as the default value instead of `nullptr` and return the result of `evaluateOrDefault` directly?

Reply:
Type() is in fact nullptr (or rather a TypeBase *nullptr wrapped in a Type())

Reply:
Either way is fine with me